static/js/components/AutoComplete.js

```diff
@@ -445,7 +454,10 @@ class AutoComplete extends Component {
       requestsList = requestsList.slice(0, maxSearchResults);
     }
 
-    this.requestsList = requestsList;
+    // 'renderItem' in Material-UI's Menu is only passed an option object when it is called downstream.
+    // This line prepares that 'renderItem' function ahead of time with some values that all MenuItems need and can't
+    // be determined by the option object alone.
+    let preparedMenuItemRenderer = _.partial(menuItemRenderer, this.getMenuItemSettings());
```

Comment:
1. even if there's only one client, i prefer this approach as it makes the component render function less cluttered and the function name expresses the intent of the block of code 
1. `renderItem` is accepted by downstream components as a prop. _that_ function is called with `option` only. the partial here is based off of `menuItemRenderer`, which is a function that takes a settings object and an option. `_.partial` satisfies the settings object, leaving a function that just takes `option`. i know it's a tad clunky. i'm thinking about some different approaches